graylog2-web-interface/src/components/common/Markdown.tsx

```diff
@@ -74,20 +81,63 @@ const useMarkdownTransformer = (): HTMLReactParserOptions['replace'] => {
   }, [components]);
 };
 
-const Augment = ({ html }: { html: string }) => {
-  const transformer = useMarkdownTransformer();
+const mergeTransformer =
+  (transformers: Array<Transformer>): Transformer =>
+  (domNode, idx) => {
+    for (const transformer of transformers) {
+      const result = transformer(domNode, idx);
+      if (result) {
+        return result;
+      }
+    }
+
+    return undefined;
+  };
+
+const Augment: Renderer = ({ html, transformer }) => {
+  const markdownTransformer = useMarkdownTransformer();
+  const replace = useMemo(
+    () => (transformer ? mergeTransformer([markdownTransformer, transformer]) : markdownTransformer),
+    [markdownTransformer, transformer],
+  );
+
+  return parse(html, { replace });
+};
+
+const UnsupportedImageWarning = () => (
+  <span>
+    Images are not supported for security reasons. Please enable allowed sources{' '}
+    <Link to={Routes.SYSTEM.configurationsSection('Markdown')}>here</Link>
+  </span>
+);
+
+const imageWarningTransformer: Transformer = (domNode) => {
+  if ('name' in domNode && domNode.name === 'img') {
+    return <UnsupportedImageWarning />;
+  }
 
-  return parse(html, { replace: transformer });
+  return undefined;
 };
 
 const Markdown = ({ augment = false, text }: Props) => {
+  const { isInitialLoading, data: markdownConfig } = useMarkdownConfig();
   // Remove dangerous HTML
   const sanitizedText = DOMPurify.sanitize(text ?? '', { USE_PROFILES: { html: false } });
 
   // Remove dangerous markdown
   const html = useMemo(() => DOMPurify.sanitize(marked(sanitizedText, { async: false })), [sanitizedText]);
+  const transformer =
+    markdownConfig?.allow_all_image_sources == true || markdownConfig?.allowed_image_sources
+      ? undefined
+      : imageWarningTransformer;
```

Comment:
It was a bit confusing for me, that we no longer show the warning after you configure the first allowed image source. Afterwards we still just show the following, for image URLs which are not part of `allowed_image_sources`.

<img width="112" height="49" alt="Image" src="https://github.com/user-attachments/assets/807de5da-9078-4155-aecf-6a5bc180197c" />

Reply:
Yes, I know. The idea was to create a follow-up testing the URL against configured sources, I did not want the scope of this PR to explode though, considering that it's already very late in the release process.